### PR TITLE
[FRE-924] feat: sei destination address evm wallet support

### PR DESCRIPTION
**Pull request by codingki**
https://github.com/user-attachments/assets/f9822f5f-e4fb-449b-b7e9-6fb83be1a60f

### CODE REVIEW

File: packages/widget/src/hooks/use-make-wallets.tsx [hunk 1]
```diff
@@ -215,6 +226,50 @@ export const useMakeWallets = () => {
...
+                  <a
+                    target="_blank"
+                    className="underline"
+                    href="https://app.sei.io/"
+                  >
+                    https://app.sei.io/
+                  </a>{' '}
+                  to associate your SEI address.
+                </p>
+              );
```
Comment (toddkao) [inline packages/widget/src/hooks/use-make-wallets.tsx:269]: @codingki can we make sure this stays open until they click on the link? also add a button to dismiss it manually

File: packages/widget/src/hooks/use-make-wallets.tsx [hunk 2]
```diff
@@ -151,9 +159,29 @@ export const useMakeWallets = () => {
...
+              }
+              return evmAddress;
+            }
+          } else {
+            await connectAsync({ connector, chainId: Number(chainID) });
+            trackWallet.track('evm', connector.id, chainType);
+
+            return evmAddress;
+          }
+        };
```
Comment (toddkao) [inline packages/widget/src/hooks/use-make-wallets.tsx:179]:
```suggestion
        const evmGetAddress: MinimalWallet['getAddress'] = async () => {
          if (connector.id !== currentConnector?.id) {
            await connectAsync({ connector, chainId: Number(chainID) });
          }
          trackWallet.track('evm', connector.id, chainType);
          return evmAddress;
        };
```

File: packages/widget/src/hooks/use-make-wallets.tsx [hunk 3]
```diff
@@ -142,7 +147,10 @@ export const useMakeWallets = () => {
       }));
     }
 
-    if (chainType === 'evm') {
+    if (
+      chainType === 'evm' ||
+      (chainType === 'cosmos' && chainID === 'pacific-1')
```
Comment (toddkao) [inline packages/widget/src/hooks/use-make-wallets.tsx:152]: I think we should make a variable for this to make it easier to read ie.
```typescript
const isSeiWallet = chainType === 'cosmos' && chainID === 'pacific-1'
if (chainType === 'evm' || isSeiWallet)

```

File: packages/widget/src/hooks/use-make-wallets.tsx (hunk 1 above)
Comment (codingki) [reply packages/widget/src/hooks/use-make-wallets.tsx:269]: yeah, will add

File: packages/widget/src/constants/abis.ts [hunk 4]
```diff
@@ -1,238 +1,272 @@
...
+    outputs: [{ internalType: 'string', name: 'response', type: 'string' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [{ internalType: 'string', name: 'addr', type: 'string' }],
+    name: 'getEvmAddr',
+    outputs: [{ internalType: 'address', name: 'response', type: 'address' }],
+    stateMutability: 'view',
+    type: 'function',
```
[the remark this answers is not in the document]
Comment (codingki) [reply packages/widget/src/constants/abis.ts:270]: I tested it

File: packages/widget/src/hooks/use-make-wallets.tsx (hunk 2 above)
Comment (codingki) [reply packages/widget/src/hooks/use-make-wallets.tsx:179]: your suggestion is always trackWallet. I'll keep the previous code

File: packages/widget/src/hooks/use-make-wallets.tsx (hunk 1 above)
Comment (codingki) [reply packages/widget/src/hooks/use-make-wallets.tsx:269]: image

File: packages/widget/src/hooks/use-make-wallets.tsx [hunk 5]
```diff
@@ -151,9 +158,30 @@ export const useMakeWallets = () => {
...
+
+              return evmAddress;
+            }
+          } else {
+            await connectAsync({ connector, chainId: Number(chainID) });
+            trackWallet.track('evm', connector.id, chainType);
+
+            return evmAddress;
+          }
+        };
```
Comment (toddkao) [inline packages/widget/src/hooks/use-make-wallets.tsx:179]:
```suggestion
        const evmGetAddress: MinimalWallet['getAddress'] = async () => {
          if (connector.id !== currentConnector?.id) {
            await connectAsync({ connector, chainId: Number(chainID) });
            trackWallet.track('evm', connector.id, chainType);
          } else if (isEvmConnected && evmAddress & signRequired) {
            trackWallet.track('evm', connector.id, chainType);
          }
          return evmAddress;
        };
```

Comment (toddkao) [reply packages/widget/src/hooks/use-make-wallets.tsx:179]: Here I added the suggested again, you should be able to click commit suggestion now that it's not outdated

**Pull request merged**